### ISSUE TITLE
Sign extension 

### ISSUE THREAD

**Issue by Hawk777** [cretz/asmble#45]
The sign extension instructions (`i32.extend8_s`, `i32.extend16_s`, `i64.extend8_s`, `i64.extend16_s`, and `i64.extend32_s`) seem like they would map quite nicely onto efficient JVM code.

**Comment by Buzzec** [comment]
These are part of the [sign-extension operators extension](https://github.com/WebAssembly/sign-extension-ops/blob/master/proposals/sign-extension-ops/Overview.md) ([opcode ref](https://pengowray.github.io/wasm-ops/)). This is now [considered default by LLVM](https://github.com/rust-lang/rust/issues/109807).

**Comment by cretz** [comment]
:+1: I don't work on this project much anymore but if/when I get around to it, that and many other extensions would be added (e.g. GC)

**Comment by Hawk777** [comment]
Not promising I’d have time or anything, but if I did, would you accept PRs?

**Comment by cretz** [comment]
Yes, but I can't promise I'd review quickly. Also, you'd have to update the test harness surely, and unsure if you'd uncover any other bugs.
